src/interface/blas3/backend/intel_gpu.hpp

```diff
@@ -92,32 +92,31 @@ typename executor_t::policy_t::event_t _gemm(
     }
   }
 #endif
-  if ((_M == 512 && _N == 49 && _K == 512) ||
-             (_M == 10 && _N == 1024 && _K == 1024)) {
-    return blas::Gemm_Launcher<
-        256, true, false, false, 64, Tile<4, 4, 16, 16>, _t_a, _t_b,
-        static_cast<int>(gemm_memory_t::local),
-        static_cast<int>(gemm_algorithm_t::classic),
-        is_beta_zero>::template _select_gemm(ex, _M, _N, _K, _alpha, _a, _lda,
-                                             _b, _ldb, _beta, _c, _ldc,
-                                             batch_size);
-  } else if (_M == 256 && _N == 784 && _K == 256) {
-    return blas::Gemm_Launcher<
-        256, false, false, false, 64, Tile<8, 8, 16, 16>, _t_a, _t_b,
-        static_cast<int>(gemm_memory_t::no_local),
-        static_cast<int>(gemm_algorithm_t::classic),
-        is_beta_zero>::template _select_gemm(ex, _M, _N, _K, _alpha, _a, _lda,
-                                             _b, _ldb, _beta, _c, _ldc,
-                                             batch_size);
-  } else if ((_M <= 128 && _N <= 128 && _K <= 128)) {
+  if (batch_size == 1 && (_t_a || (_t_b && _M * _N > 1048576))) {
+    if (_M <= 64 || _N <= 64) {
+      return blas::Gemm_Launcher<
+          64, true, true, true, 64, Tile<4, 4, 8, 8>, _t_a, _t_b,
+          static_cast<int>(gemm_memory_t::local),
+          static_cast<int>(gemm_algorithm_t::alternative),
+          is_beta_zero>::template _select_gemm(ex, _M, _N, _K, _alpha, _a, _lda,
+                                               _b, _ldb, _beta, _c, _ldc, 1);
+    } else {
+      return blas::Gemm_Launcher<
+          256, true, true, true, 64, Tile<4, 4, 16, 16>, _t_a, _t_b,
+          static_cast<int>(gemm_memory_t::local),
+          static_cast<int>(gemm_algorithm_t::alternative),
+          is_beta_zero>::template _select_gemm(ex, _M, _N, _K, _alpha, _a, _lda,
+                                               _b, _ldb, _beta, _c, _ldc, 1);
+    }
+  } else if (_M <= 128 && _N <= 128) {
     return blas::Gemm_Launcher<
         64, true, false, false, 64, Tile<4, 4, 8, 8>, _t_a, _t_b,
         static_cast<int>(gemm_memory_t::local),
         static_cast<int>(gemm_algorithm_t::classic),
         is_beta_zero>::template _select_gemm(ex, _M, _N, _K, _alpha, _a, _lda,
                                              _b, _ldb, _beta, _c, _ldc,
                                              batch_size);
-  } else if (!_t_a) {
+  } else if (_t_b && !_t_a) {
```

Comment:
Could this be changed to _t_b != _t_a or is the performance significantly different when b is not transposed and a is?

Reply:
It really can't. The algorithm without local memory is better than the algorithm with local memory only in the case when both of the matrices are naturally accessed in a coalesced pattern. With column-major matrices this is when B is transposed and A isn't.

Changing that to `_t_b != _t_a` would give awful performance in the case where A is transposed and B isn't (non-coalesced access in both).

Reply:
OK no worries, thanks for clarifying that!